### PR TITLE
Prevents cult teleport while prone

### PR DESCRIPTION
**Pull request by Coolrune206**
## What Does This PR Do

As it states on the tin, if you are on the ground for any reason and you try to cast the cult teleport spell, you will be unable to.
## Why It's Good For The Game

Discussing cult balance is a bit of a joke in the community, but I think most will agree that cult at the moment is an exceptionally safe antag for a number of reasons - they can be fully rejuvenated if they die using a revive rune, they can be recalled to a recall rune by their fellow cultists, and at any point they feel like they can use the teleport spell to escape any ill situation. With this change at least, if you baton a cultist they cannot immediately cast this spell to get out risk-free from any combat.
## Testing

Made sure it worked for both stuns, knockdowns, and intentional resting, has a message to the cultist that it doesn't work while prone and doesn't use up the spell.
## Changelog
:cl:
tweak: Cultists are now unable to use the teleport spell while prone
/:cl:

### CODE REVIEW

**Comment by Samman166** [comment]
I have mixed feelings on this, on one hand I think cult having a lot of recovery options is fine because it stops cultstended, but then when you apply that to when cult gets bigger it means sec can barely take out any cultists. One glaring issue I see is it will make the skill gap even more clear, robust players will be able to react to a baton hit quickly and use the teleport before knocking down, while it punishes newer players who don't have that reaction times. However I guess that is just the eternal dilemma of cult balance. Granted I'm probably just over analyzing since before stamina combat one baton stun was enough to stop a cultist from teleporting and cult balance was still wack back then.

**Comment by Komrad822** [comment]
My main issue with this is teleport spell is already droppable, as Samman said this makes it a skill check if the person got batoned. The same seccie doesnt even have to baton them again, they can either have someone else baton the same person and then detain them or disabler them down after the second hit, at which point they are stunned and cant teleport themselves anyway. A cultist can be made horizontal by water, glass without shoes, shoved into a wall, tables, lucky weaken hit, baton, telebaton, disabler, being in crit, magic missile, explosion, bsg, meme chemed, addictions, flash bang, flashed, ruber shot, bean bag, terror web, xeno disarm, honk mech and krav gloves. Of those that I listed (prob missing quite a few), those only applied by sec which usually end up in a stun, should stop cult from teleporting away.
There is a reason we get cultstended and its because cult is already quite hard for new players. Yes there are robustos, yes sec gets their shit kicked in sometimes but its a dynamic team game mode that transforms throughout the round. We always complain about cult balance when cult wins but never when they lose. Why did cult lose? We say skill issue. Why did sec lose? We say cult is unbalanced. But maybe it was because command doesnt just force shuttle call when they notice they've lost two officers of their four sec officer roster or dont immediately call ERT AND mind shield all command members.

Revive runes require two mindshielded people to have been sacrificed per one revive, which uses up those two in the pool.
Recall rune requires the target to not be held or buckled and requires two people to do the recall. If off station, the user cannot summon a ghost to assist.
Teleport requires you to have the spell and not drop it. As I said before there is more that one situation that will make the cultist horizontal. 

If we are going to make teleport not work while knocked down then we should make it no drop because loosing it because you HAVE to stand up while being shove spammed, pushed on space lube, lucky weaken hit, or fighting crit blackouts is just as unfun for cultist as it is any other antag. 

TLDR: Theres to many situations for a cultist to be made horizontal other than those applied by sec. Cult already an enemy of the crew, command, and CC. Valid hunting is prevalent and cultstended happens a lot. Removing the instances spells can be used just punishes people because others are bald and dont know how to properly handle someone when they have the upper hand.

**Comment by Komrad822** [comment]
Or add a like three minute timer before you can use the teleport spell again. Thus making it so they cant immediately get safe and then put themselves into a risky situation again without consequence, just like every other form of player initiated teleport spell (jaunt, blink, vamp abilities old and new, you get sick with bluespace teleports).

**Review by Sirryan2002 — APPROVED** [review]
code looks good

File: code/game/gamemodes/cult/blood_magic.dm [hunk 1]
```diff
@@ -515,6 +515,10 @@
 		return
 	uses--
 
+	if(IS_HORIZONTAL(user))
+		to_chat(user, "<span class='cultitalic'>You cannot cast this spell while on the floor!</span>")
+		return
```
Comment (hal9000PR) [inline code/game/gamemodes/cult/blood_magic.dm:520]: would you make this check for the floored trait instead.

File: code/game/gamemodes/cult/blood_magic.dm [hunk 2]
```diff
@@ -515,6 +515,10 @@
...
+
```
[the remark this answers is not in the document]
Comment (Coolrune206) [reply code/game/gamemodes/cult/blood_magic.dm:521]: that would probably make sense yes

**Review by Sirryan2002 — APPROVED** [review]
